bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdConfigFileChecker.java

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Bachmann electronic GmbH and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.internal.clangd;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.eclipse.cdt.lsp.internal.clangd.editor.ClangdPlugin;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Checks the <code>.clangd</code> file for syntax errors and notifies the user via error markers in the file and Problems view.
+ */
+public class ClangdConfigFileChecker {
+	public static final String CLANGD_MARKER = ClangdPlugin.PLUGIN_ID + ".config.marker"; //$NON-NLS-1$
+	private final Pattern pattern = Pattern.compile(".*line (\\d+), column (\\d+).*"); //$NON-NLS-1$
+	private boolean temporaryLoadedFile = false;
+
+	/**
+	 * Checks if the .clangd file contains valid yaml syntax. Adds error marker to the file if not.
+	 * @param configFile
+	 */
+	public void checkConfigFile(IFile configFile) {
+		Yaml yaml = new Yaml();
+		try (var inputStream = configFile.getContents()) {
+			try {
+				removeMarkerFromClangdConfig(configFile);
+				//throws ScannerException and ParserException:
+				yaml.load(inputStream);
+			} catch (Exception yamlException) {
+				addMarkerToClangdConfig(configFile, yamlException);
```

Comment:
Many Yaml exceptions - like the one raised in https://github.com/eclipse-cdt/cdt-lsp/issues/245#issue-2119111356 -  are of type [`MarkedYAMLException`](https://www.javadoc.io/doc/org.yaml/snakeyaml/2.0/org/yaml/snakeyaml/error/MarkedYAMLException.html) which have some utility methods like [`getProblemMark`](https://www.javadoc.io/static/org.yaml/snakeyaml/2.0/org/yaml/snakeyaml/error/MarkedYAMLException.html#getProblemMark()) and [`getProblem`](https://www.javadoc.io/static/org.yaml/snakeyaml/2.0/org/yaml/snakeyaml/error/MarkedYAMLException.html#getProblem()) which will allow you to deconstruct the exception without having to parse the string.

I haven't tried to use them in this context, but it looks like it should do the job

Reply:
Thank you for hint! It's much easier!